openwisp_controller/config/apps.py

```diff
@@ -172,6 +175,31 @@ def enable_cache_invalidation(self):
             DeviceChecksumView.invalidate_checksum_cache,
             dispatch_uid='invalidate_checksum_cache',
         )
+        post_save.connect(
+            devicegroup_change_handler,
+            sender=self.devicegroup_model,
+            dispatch_uid='invalidate_devicegroup_cache_on_devicegroup_change',
+        )
+        post_save.connect(
+            devicegroup_change_handler,
+            sender=self.org_model,
+            dispatch_uid='invalidate_devicegroup_cache_on_organization_change',
+        )
+        post_save.connect(
+            devicegroup_change_handler,
+            sender=self.cert_model,
+            dispatch_uid='invalidate_devicegroup_cache_on_certificate_change',
+        )
+        pre_delete.connect(
```

Comment:
let's add a comment here mentioning we have to use pre_delete because we need the data to be present, is that the case right?

Reply:
With some changes in other parts of the code, I have made it to post_delete as well.